Sources/KituraNet/ClientRequest.swift

```diff
@@ -550,8 +551,20 @@ public class ClientRequest {
             initializeClientBootstrap(eventLoopGroup: group)
         }
 
-        let hostName = URL(string: percentEncodedURL)?.host ?? "" //TODO: what could be the failure path here
-        let portNumber = URL(string: percentEncodedURL)?.port ?? 8080
+        let hostName = percentEncodedURL.host! //TODO: what could be the failure path here
+        let portNumber: Int
+        if let port = percentEncodedURL.port {
+            portNumber = port
+        } else {
+            if percentEncodedURL.scheme == "https" {
+                portNumber = 443
+            } else if percentEncodedURL.scheme == "http" {
+                portNumber = 80
+            } else {
+                portNumber = 8080
+                Log.error("Unknown port for url: \(url)  Assuming 8080")
```

Comment:
Well, technically it's not an error, since you have recovered in a transparent & seamless way.
I'd suggest implementing it as a warning instead of an error.

Reply:
That's true... but at the same time I don't think having a default 8080 port makes any sense.  I kept it there to avoid introducing more error paths and since the previous code had it.  You think it should be a warning instead of an error?